mlflow/projects/utils.py

```diff
@@ -362,4 +362,9 @@ def get_databricks_env_vars(tracking_uri):
         env_vars["DATABRICKS_TOKEN"] = config.token
     if config.ignore_tls_verification:
         env_vars["DATABRICKS_INSECURE"] = str(config.ignore_tls_verification)
+
+    workspace_info = databricks_utils.get_databricks_workspace_info_from_uri(tracking_uri)
+    if workspace_info is not None:
+        env_vars.update(workspace_info.to_environment())
```

Comment:
Introduce two new internal env vars (`_DATABRICKS_WORKSPACE_HOST` and `_DATABRICKS_WORKSPACE_ID`) and set them in the subprocess used to run the train / evaluate / register steps because subprocesses cannot read this information from the command context on Databricks (command context isn't available to subprocesses).